src/NHapi.Base/Parser/ParserBase.cs

```diff
@@ -121,13 +112,13 @@ public static ISegment MakeControlMSH(string version, IModelClassFactory factory
                     (IMessage)GenericMessage
                         .GetGenericMessageClass(version)
                         .GetConstructor(new[] { typeof(IModelClassFactory) })
-                        .Invoke(new object[] { factory });
+                        ?.Invoke(new object[] { factory });
 
                 var c = factory.GetSegmentClass("MSH", version);
                 var constructorParamTypes = new[] { typeof(IGroup), typeof(IModelClassFactory) };
                 var constructorParamArgs = new object[] { dummy, factory };
                 var constructor = c.GetConstructor(constructorParamTypes);
-                msh = (ISegment)constructor.Invoke(constructorParamArgs);
+                msh = (ISegment)constructor?.Invoke(constructorParamArgs);
```

Comment:
this would have thrown previously, now it result in returning null I think when `constructor` is null?

Reply:
You are correct, will review if this makes a difference.

Reply:
will change to `!.` if this does blow up as could be null it is already wrapped in try catch like it was before.